src/main/java/fr/sncf/osrd/infra_state/RouteState.java

```diff
@@ -121,11 +140,29 @@ public void reserve(Simulation sim) throws SimulationError {
         sim.publishChange(change);
         notifySignals(sim);
 
-        // Reserve the tvd sections
-        for (var tvdSectionPath : route.tvdSectionsPaths) {
-            var tvdSection = sim.infraState.getTvdSectionState(tvdSectionPath.tvdSection.index);
-            tvdSection.reserve(sim);
+        reserveTvdSection(sim);
+    }
+
+    /**
+     * Reserve a route in cbtc and his tvd sections. Routes that share tvd sections
+     * will have the status CONFLICT
+     */
+    public void cbtcReserve(Simulation sim) throws SimulationError {
+        if (status != RouteStatus.CBTC_OCCUPIED && status != RouteStatus.CBTC_RESERVED
+                && status != RouteStatus.CBTC_REQUESTED) {
+            return;
         }
```

Comment:
Be careful about this block. It seems those status are only set in this function, so we will always be false. Why is it needed?